docs/functions/utility.md

```diff
@@ -1,5 +1,47 @@
 # Utility functions
 
+## deepOverwriteMerge
+
+```typescript
+deepOverwriteMerge<T extends object, U extends object>(
+	target: T,
+	overrides?: U,
+): T & U
```

Comment:
Here's a nitpick I am pondering. What's the best developer experience for the type argument naming here?
* `T, U` because that's concise and often used for generics with 2 arguments
* `T, O`, which are still concise but they line up with the parameter names instead of being abstract
* `TTarget` and `TOverrides`, which is less concise but more clear, I see this in the wild sometimes